datafusion/core/src/physical_plan/joins/sort_merge_join.rs

```diff
@@ -249,6 +249,17 @@ impl ExecutionPlan for SortMergeJoinExec {
         self.output_ordering.as_deref()
     }
 
+    fn maintains_input_order(&self) -> Vec<bool> {
+        match self.join_type {
+            JoinType::Inner => vec![true, true],
+            JoinType::Left | JoinType::LeftSemi | JoinType::LeftAnti => vec![true, false],
```

Comment:
the `false` here is because NULLs can be injected into the stream arbitrarily, right, messing up the sort order?

Reply:
Exactly.

Reply:
👍